xcm/src/v0/junction.rs

```diff
@@ -32,6 +32,41 @@ pub enum NetworkId {
 	Kusama,
 }
 
+/// An identifier of a pluralistic body.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode, Debug)]
+pub enum BodyId {
+	/// The only body in its context.
+	Unit,
+	/// A named body.
+	Named(Vec<u8>),
+	/// An indexed body.
+	// TODO: parity-scale-codec#262: Change to be a tuple.
+	Index { #[codec(compact)] id: u32 },
+	/// The unambiguous executive body (for Polkadot, this would be the Polkadot council).
+	Executive,
+	/// The unambiguous technical body (for Polkadot, this would be the Technical Committee).
+	Technical,
+	/// The unambiguous legislative body (for Polkadot, this could be considered the opinion of a majority of
+	/// lock-voters).
+	Legislative,
+	/// The unambiguous judicial body (this doesn't exist on Polkadot, but if it were to get a "grand oracle", it
+	/// may be considered as that).
+	Judicial,
+}
+
+/// A part of a pluralistic body.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode, Debug)]
+pub enum BodyPart {
+	/// The body's declaration, under whatever means it decides.
+	Voice,
```

Comment:
@gavofyork Can you elaborate on this? Does this mean it could be any of [one member, a number of members, half the body, the full body]? What's the mapping to the e.g. the collective or democracy origins.

Reply:
Collective wouldn't have a mapping since there's no unambiguous way for it to make a declaration.

Democracy would be the already established way - in Polkadot Relay chain, this would be a lock-vote with the AQB determined by council support.